aten/src/ATen/native/native_functions.yaml

```diff
@@ -4056,6 +4056,10 @@
   device_check: NoCheck
   device_guard: False
 
+- func: _reshape_nested(Tensor self, int[] shape) -> Tensor
```

Comment:
Why is this schema different from the `reshape` above?
This function is also return sometimes a view and sometimes not a view.

So this schema is wrong (and autograd is going to generate silently wrong code for it!)

It is NOT possible to register autograd for a function that has such a property so you will need to keep reshape CompositImplicitAutograd.
Since the `computeStride` function there will not work with nested, you can simply create a new native function to abstract it away: You can make it a native function that takes `self` as input directly and you will be able to provide a custom Nested implementation for it.

Reply:
btw this wrong schema is also most likely the reason for the inference mode issue you had as the autograd kernel is not properly registered!